.github/workflows/release.yaml

```diff
@@ -56,10 +56,110 @@ jobs:
         destination: tanzu-cli-admin-plugins
         credentials: ${{ secrets.GCP_BUCKET_SA }}
 
+    - name: Generate details for creating release notes
+      id: generate-release-details
+      shell: bash
+      run: |
+        set -o xtrace
+        RELEASE_VERSION=${GITHUB_REF#refs/*/}
+        major=$(echo "$RELEASE_VERSION"  | sed 's/^v\(.*\)/\1/' | cut -d. -f1)
+        minor=$(echo "$RELEASE_VERSION" | cut -d. -f2)
+        revision=$(echo "$RELEASE_VERSION" | cut -d. -f3)
+        if [ "$revision" -gt 0 ];then
+          revision=$(($revision-1))
+        elif [ "$minor" -gt 0 ]; then
+          minor=$(($minor-1))
+        elif [ "$major" -gt 0 ]; then
+          major=$(($major-1))
+        else
+          echo "Please validate that the tag release version(${RELEASE_VERSION}) conforms to semver."
+          exit 1
+        fi
+        LAST_TAG="v${major}.${minor}.${revision}"
+        echo "Last release tag - $LAST_TAG"
+        START_SHA=$(git rev-list -n 1 $LAST_TAG)
+        echo "Release note generator start SHA - $START_SHA"
+        END_SHA=$(git rev-list -n 1 $RELEASE_VERSION)
+        echo "Release note generator end SHA - $END_SHA"
+
+        echo "::set-output name=start-sha::$START_SHA"
+        echo "::set-output name=end-sha::$END_SHA"
+        echo "::set-output name=release-version::$RELEASE_VERSION"
+        echo "::set-output name=prev-release-version::$LAST_TAG"
+        echo "::set-output name=repo-name::$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')"
+        echo "::set-output name=repo-org::$(echo '${{ github.repository }}' | awk -F '/' '{print $1}')"
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Install release-note dependency
+      run: |
+        go get k8s.io/release/cmd/release-notes@latest
```

Comment:
I am wondering if we should pin the sha which works now instead of `latest` for the sake of reproducibility.

Reply:
The reason I am not pinning to SHA is simply because other wise we will need to regularly update the workflow every time a new release notes generator version is released.

Reply:
ack! The only issue with release-notes@latest is that if release-notes introduces a different ux in the tool then our workflow will break. I think we go ahead with latest for now because ux changing is a less likely scenario